lib/irb/init.rb

```diff
@@ -244,7 +245,8 @@ def IRB.init_error
   end
 
   # option analyzing
-  def IRB.parse_opts(argv: ::ARGV)
+  def IRB.parse_opts
+    argv = @CONF[:ARGV].dup || []
```

Comment:
Problems in tests:

1. https://github.com/ruby/irb/blob/a641746b18faa35ef32eec55255894fffcd4b063/test/irb/test_init.rb#L179
2. https://github.com/ruby/irb/blob/a641746b18faa35ef32eec55255894fffcd4b063/test/irb/test_init.rb#L206

Both of these assert that the ARGV array passed as argument is mutated. Is this really a behaviour that is important? Isn't it better to assert that the IRB.conf[...] values are properly set?